evalml/automl/automl_search.py

```diff
@@ -150,21 +150,6 @@ def search(
     y_train = infer_feature_types(y_train)
     problem_type = handle_problem_types(problem_type)
 
-    datetime_column = None
```

Comment:
Removing this because `DefaultDataChecks` are run anyways which includes `TimeSeriesParametersDataCheck`